src/publicapi.ts

```
@@ -545,24 +576,26 @@ window.MathQuill = MathQuill;
 
 function RootBlockMixin(_: RootBlockMixinInput) {
   _.moveOutOf = function (dir: Direction) {
-    this.controller.handle('moveOutOf', dir);
+    this.controller?.handle('moveOutOf', dir);
```

What's the case where `controller` can be undefined? Is that a thing that only happens in earlier interface versions?

---

In practice I think it's never undefined, but the the way the types are set up, that fact is not guaranteed at the type level

---

In that case, it might be better to use `!` instead of `?` here so that we get runtime errors instead of silent errors in case this assumption is incorrect.